pandas/tests/extension/test_arrow.py

```diff
@@ -1937,10 +1942,24 @@ def test_str_find_negative_start():
     tm.assert_series_equal(result, expected)
 
 
-def test_str_find_notimplemented():
+def test_str_find_no_end():
     ser = pd.Series(["abc", None], dtype=ArrowDtype(pa.string()))
-    with pytest.raises(NotImplementedError, match="find not implemented"):
-        ser.str.find("ab", start=1)
+    if pa_version_under13p0:
+        # https://github.com/apache/arrow/issues/36311
+        with pytest.raises(pa.lib.ArrowInvalid, match="Negative buffer resize"):
+            ser.str.find("ab", start=1)
+    else:
+        result = ser.str.find("ab", start=1)
+        expected = pd.Series([-1, None], dtype="int64[pyarrow]")
+        tm.assert_series_equal(result, expected)
+
+
+def test_str_find_negative_start_negative_end():
+    # GH 56791
+    ser = pd.Series(["abcdefg", None], dtype=ArrowDtype(pa.string()))
+    result = ser.str.find(sub="d", start=-6, end=-3)
```

Comment:
1. What happens when `start` would wrap around a string entry e.g. `start=16` here
2. What would happen if `start=-3, end=-6`?

Reply:
Looking at code again, this approach doesn't handle case where sub is empty string correctly. I think a different approach should be used that avoids substring, going to mark as a draft, and refactor, and add tests for empty string case.

Reply:
I've adjusted the code to handle empty strings as the substring, I also added test_str_find_e2e, that enumerates multiple combinations and baselines against the python str.find, I believe and e2e test is the best way to test this, manually picking a few cases is unlikely to cover all cases for this API, there are too many interesting cases.